src/pages/Login.js

```diff
@@ -42,35 +42,30 @@ function Login() {
   });
 
   const onSubmit = (formData) => {
-    if (!formData) {
-      return;
-    } else {
+    if (formData) {
       run(authApi.login(formData.email, formData.password));
     }
   };
   useEffect(() => {
     if (data?.token) {
       profileRun(profileApi.myProfile());
     }
-  }, [data, profileData, profileRun, setAuthState]);
+  }, [data?.token, profileRun]);
+
+  useEffect(() => {
+    if (data?.token || profileData) {
+      setAuthState({ token: data?.token, user: profileData });
+    }
+  }, [data?.token, profileData, setAuthState]);
 
   if (status === 'pending') {
```

Comment:
make sure to have the spinner rendering, when either token request or profile request is still pending